File: Assets/Scripts/Enemies/EnemyManager.js
import UnityEngine.UI;

var playerHealth : PlayerHealth;       // Reference to the player's heatlh.
var enemy : GameObject;                // The enemy prefab to be spawned.
var spawnTime : float = 8f;            // How long between each spawn.
var spawnPoints : Transform[];         // An array of the spawn points this enemy can spawn from.
var quantity= 10;
public var timerSeconds = 10;
var enemiesHaveSpawn = false;


function Start ()
{
    // Call the Spawn function after a delay of the spawnTime and then continue to call after the same amount of time.
    //InvokeRepeating ("Spawn", spawnTime, spawnTime);
}
/**
function Update(){

if(GameObject.FindGameObjectsWithTag("Enemy").Length == 10){
	CancelInvoke();
}
}**/

function Spawn (spawnPointIndex: int)
{
    // If the player has no health left...
    if(playerHealth.Health <= 0f)
    {
        // ... exit the function.
        return;
    }

    // Find a random index between zero and one less than the number of spawn points.
    
    // Create an instance of the enemy prefab at the randomly selected spawn point's position and rotation.
    Instantiate (enemy, spawnPoints[spawnPointIndex].position, spawnPoints[spawnPointIndex].rotation);
    Debug.Log("Instatiate enemy");
    
}
function CountDown(){
 
   timerSeconds--;
 
   if(timerSeconds < 1){
 
     print ("Count down finished");
 
     CancelInvoke("Spawn");
 
   }
 
 }

function OnTriggerEnter (player : Collider) {
				if(player.tag == "Player" && !enemiesHaveSpawn){
			for (i=0; i < quantity; i++){
				var enemySpawnPoint = i%spawnPoints.Length;
				Spawn(enemySpawnPoint);
				yield WaitForSeconds(5);
			}
			enemiesHaveSpawn = true;
		}
		Debug.Log("Spawnie mis enemigos");
	}

File: Assets/Scripts/ULTIMATE/Enemies/EnemyManager.cs
using UnityEngine;
using System.Collections;

public class EnemyManager : MonoBehaviour {
	
	private PlayerHealth playerHealth;       // Reference to the player's heatlh.
	public GameObject enemy;                // The enemy prefab to be spawned.
	public float spawnTime= 8f;            // How long between each spawn.
	public Transform[] spawnPoints;         // An array of the spawn points this enemy can spawn from.
	public int quantity= 10;
	public int timerSeconds = 5;
	private bool enemiesHaveSpawn = false;
	private bool spawnEnemies = false;
	
	// Use this for initialization
	void Awake () {
		GameObject player = GameObject.FindGameObjectWithTag ("Player");
		if (player != null) {
			playerHealth = (PlayerHealth)player.GetComponent<PlayerHealth> ();
		}
	}
	
	// Update is called once per frame
	void Update () {
		if (spawnEnemies && !enemiesHaveSpawn) {
			if(playerHealth!=null && playerHealth.currentHealth > 0 && spawnPoints.Length > 0)
			{
				for (int i=1; i < quantity; i++){
					int enemySpawnPoint = i%spawnPoints.Length;
					StartCoroutine("SpawnEnemies",enemySpawnPoint);	
				}
				enemiesHaveSpawn = true;
			}
		}
	}

	IEnumerator SpawnEnemies (int enemySpawnPoint)
	{

		// Create an instance of the enemy prefab at the randomly selected spawn point's position and rotation.
		Instantiate (enemy, spawnPoints[enemySpawnPoint].position, spawnPoints[enemySpawnPoint].rotation);
		waitToSpawn();

		// If the player has no health left...
		yield return null;
	}
	
	void OnTriggerEnter (Collider player) {
		if(player.tag == "Player" && !enemiesHaveSpawn){
			spawnEnemies = true;
		}
		Debug.Log("Spawnie mis enemigos");
	}
	
	IEnumerable waitToSpawn(){
		yield return new WaitForSeconds(timerSeconds);
	}
}
